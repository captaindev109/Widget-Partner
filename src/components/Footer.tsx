import { FC, ReactNode } from 'react';
import Card from '@/generics/Card';

export interface FooterProps {
  padding?: string;
  background?: string;
  shadow?: number | string;
  children?: ReactNode;
}

const Footer: FC<FooterProps> = (props) => {
  return <Card {...props} width="100%" position="absolute" bottom="0px" data-testid="footer" />;
};

Footer.defaultProps = {
  padding: '16px',
  background: 'white',
  shadow: 'footer',
};

export default Footer;
