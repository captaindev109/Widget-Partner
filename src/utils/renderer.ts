import React from 'react';
import { ConfigProps, Index, EachComponentObjectChildren, BasicAny } from '@/types/types';
import { UIComponents } from './components';

const recursivelyCreateReactComponents = (
  componentToCreate: BasicAny,
  configuration: ConfigProps,
  children: BasicAny
) => {
  if (typeof componentToCreate !== 'undefined') {
    return React.createElement(
      componentToCreate,
      configuration,
      children &&
        (typeof children === 'string'
          ? children
          : children.map((component: EachComponentObjectChildren, index: Index) => viewRenderer(component, index)))
    );
  }
  null;
};

const viewRenderer: BasicAny = (config: ConfigProps, index: Index) => {
  const children = config.children;
  const newConfig = { ...config, key: index };
  const LookForComponentInUILibrary = UIComponents[config.component];
  return recursivelyCreateReactComponents(LookForComponentInUILibrary, newConfig, children);
};

export default viewRenderer;
