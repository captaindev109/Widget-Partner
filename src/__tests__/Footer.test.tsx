import { render, RenderResult } from '@testing-library/react';
import queries from '@testing-library/dom/types/queries';
import renderer from 'react-test-renderer';
import Footer from '@/components/Footer';

const children = 'footer';

const STYLE_PROPS: Record<string, string> = {
  padding: '10px',
  background: '#121212',
};

/** Render tests */

describe('Footer render tests', () => {
  let renderedComponent: RenderResult<typeof queries, HTMLElement>;

  beforeEach(() => {
    renderedComponent = render(<Footer>{children}</Footer>);
  });

  test('to see if the component render without crashing', () => {
    render(<Footer>{children}</Footer>);
  });

  test('to see if the footer renders correctly', () => {
    const { getByTestId } = renderedComponent;
    const footer = getByTestId('footer');

    expect(footer).toBeTruthy();
    expect(footer.textContent).toBe(children);
  });
});

/** Props tests */

describe('Footer props test', () => {
  let renderedComponent: RenderResult<typeof queries, HTMLElement>;

  beforeEach(() => {
    const props = { ...STYLE_PROPS };
    renderedComponent = render(<Footer {...props}>{children}</Footer>);
  });

  test('to see if the footer has all the necessary styling', () => {
    const { getByTestId } = renderedComponent;
    const footer = getByTestId('footer');

    // Mapping through the object to check if the style props are applied
    Object.entries(STYLE_PROPS).forEach(([key, value]) => {
      expect(footer).toHaveStyle(`${key}: ${value}`);
    });
    expect(footer).toHaveStyle(`bottom: 0`);
    expect(footer).toHaveStyle(`position: absolute`);
  });

  it('Snapshot test should pass', () => {
    const snap = renderer.create(<Footer>{children}</Footer>).toJSON();

    expect(snap).toMatchSnapshot();
  });
});
