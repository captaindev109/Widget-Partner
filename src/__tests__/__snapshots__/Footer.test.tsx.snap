// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer props test Snapshot test should pass 1`] = `
<div
  className="Card-sc-fb64gk-0 cOsjRF"
  data-testid="footer"
  width="100%"
>
  footer
</div>
`;
